web/controllers/users_controller.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Notox.UserController do
+  use Notox.Web, :controller
+  alias Notox.User
+
+  def index(conn, _params) do
+    users = Repo.all(User)
+    render conn, :index, users: users
+  end
+
+  def show(conn, %{"id" => id}) do
+    user = Repo.get(User, id)
+    render conn, :show, user: user
+  end
+
+  def new(conn, _params) do
+    changeset = User.changeset(%User{})
+    render conn, :new, changeset: changeset
+  end
+
+  def create(conn, %{"user" => user_params}) do
+    changeset = User.registration_changeset(%User{}, user_params)
+    case Repo.insert(changeset) do
+      {:ok, user} ->
+        conn
+        |> Notox.Auth.login(user)
+        |> put_flash(:info, "User created!")
+        |> redirect(to: user_path(conn, :index))
+      {:error, changeset} ->
+        render(conn, :new, changeset: changeset)
+    end
+  end
+
+  def edit(conn, %{"id" => id}) do
```

Comment:
I didn't get to restricting access yet. Wouldn't this fall under that category. I'm imagining checking the `id` against `get_session(conn, :user_id)` to make sure this user can only be edited when this user is logged in.